utils/module_synopsis.sh

```diff
@@ -18,43 +18,42 @@
 #   (TeX needs to be able to find grasslogo_vector.pdf)
 #
 
-if  [ -z "$GISBASE" ] ; then
+if [ -z "$GISBASE" ]; then
     echo "You must be in GRASS GIS to run this program." 1>&2
     exit 1
 fi
 
-for FILE in txt tex ; do
-    if [ -e "$GISBASE/etc/module_synopsis.$FILE" ] ; then
-    #  echo "ERROR: module_synopsis.$FILE already exists" 1>&2
-    #  exit 1
-       #blank it
-       g.message -w "Overwriting \"\$GISBASE/etc/module_synopsis.$FILE\""
-       \rm "$GISBASE/etc/module_synopsis.$FILE"
+for FILE in txt tex; do
+    if [ -e "$GISBASE/etc/module_synopsis.$FILE" ]; then
+        #  echo "ERROR: module_synopsis.$FILE already exists" 1>&2
+        #  exit 1
+        #blank it
+        g.message -w "Overwriting \"\$GISBASE/etc/module_synopsis.$FILE\""
+        \rm "$GISBASE/etc/module_synopsis.$FILE"
     fi
 done
-for FILE in html pdf ; do
-    if [ -e "$GISBASE/docs/$FILE/module_synopsis.$FILE" ] ; then
-    #  echo "ERROR: module_synopsis.$FILE already exists" 1>&2
-    #  exit 1
-       #blank it
-       g.message -w "Overwriting \"\$GISBASE/docs/$FILE/module_synopsis.$FILE\""
-       \rm "$GISBASE/docs/$FILE/module_synopsis.$FILE"
+for FILE in html pdf; do
+    if [ -e "$GISBASE/docs/$FILE/module_synopsis.$FILE" ]; then
+        #  echo "ERROR: module_synopsis.$FILE already exists" 1>&2
+        #  exit 1
+        #blank it
+        g.message -w "Overwriting \"\$GISBASE/docs/$FILE/module_synopsis.$FILE\""
+        \rm "$GISBASE/docs/$FILE/module_synopsis.$FILE"
     fi
 done
 
-
-TMP="`g.tempfile pid=$$`"
-if [ $? -ne 0 ] || [ -z "$TMP" ] ; then
+if ! TMP="$(g.tempfile pid=$$)" || [ -z "$TMP" ]; then
     g.message -e "Unable to create temporary files"
     exit 1
 fi
 
-g.message "Generating module synopsis (writing to \$GISBASE/etc/) ..."
+# shellcheck disable=SC2016 # Expressions don't expand in single quotes.
+g.message 'Generating module synopsis (writing to $GISBASE/etc/) ...'
```

Comment:
No need to change anything, revert to original.

Reply:
If a string doesn't need to expand variables, it can use single quotes.
I wanted to explicitly show that we wanted the $GISBASE as an unevaluated string.

Reply:
> I wanted to explicitly show that we wanted the $GISBASE as an unevaluated string.

That's what the original code does with `\`.

I really don't understand why we would change a valid and clear code to something that needs to be suppressed.

Reply:
We want code where both machine and human will understand what the author meant ideally without a comment (for human or machine). Does the original version generate a warning?